File: tests/fixtures/header_injection/python/unsafe_set_header.py
# Unsafe: Flask response.headers.add receives a value built from request
# args.  HEADER_INJECTION fires on the value argument.
from flask import request, make_response


def handler():
    lang = request.args.get("lang")
    resp = make_response("ok")
    resp.headers.add("X-Lang", lang)
    return resp

File: tests/fixtures/header_injection/go/unsafe_set_header.go
// Unsafe: net/http `ResponseWriter.Header().Set` receives a value built from
// `r.URL.Query().Get`.  HEADER_INJECTION fires on the value argument.
package main

import (
	"net/http"
)

func handler(w http.ResponseWriter, r *http.Request) {
	lang := r.URL.Query().Get("lang")
	w.Header().Set("X-Lang", lang)
}
